ReSharper.FSharp/src/FSharp.Psi.Features/src/Injected/FSharpLiteralInjectionTarget.fs

```diff
@@ -0,0 +1,101 @@
+        | _ -> ()
+
+    interface IInjectionTargetLanguage with
+        override _.ShouldInjectByAnnotation(_, prefix, postfix) =
+            prefix <- null
+            postfix <- null
+            false
+
+        override _.GetStartOffsetForString(originalNode) =
+            originalNode.GetTokenType() |> getStringStartingQuotesLength
+
+        override _.GetEndOffsetForString(originalNode) =
+            originalNode.GetTokenType() |> getStringEndingQuotesLength
+
+        override _.UpdateNode(_, _, _, length, _, _, _, _) =
+            length <- -1
+            null
+
+        override _.SupportsRegeneration = false
+
+        override _.IsInjectionAllowed(node) =
+            tryOpenChameleon node
+
+            let tokenType = node.GetTokenType()
+
+            FSharpTokenType.Strings[tokenType] &&
+            tokenType <> FSharpTokenType.CHARACTER_LITERAL &&
+            tokenType <> FSharpTokenType.VERBATIM_BYTEARRAY &&
+            tokenType <> FSharpTokenType.BYTEARRAY
+
+        override _.GetCorrespondingCommentTextForLiteral _ = null
+
+        override _.CreateBuffer(literalNode, text, options) =
+            let literalType =
+                match literalNode.GetTokenType().GetLiteralType() with
+                | RegularString ->
+                    CSharpLiteralType.RegularString :> CSharpLiteralType
+                | InterpolatedString ->
+                    CSharpLiteralType.RegularInterpolatedString
+                | InterpolatedStringStart ->
+                    CSharpLiteralType.RegularInterpolatedStringStart
+                | InterpolatedStringMiddle ->
+                    CSharpLiteralType.RegularInterpolatedStringMiddle
+                | InterpolatedStringEnd ->
+                    CSharpLiteralType.RegularInterpolatedStringEnd
+                | _ ->
+                    CSharpLiteralType.VerbatimString
+
+            let lexerOptions =
+                match options with
+                | :? RegexOptions as regexOptions when regexOptions.HasFlag(RegexOptions.IgnorePatternWhitespace) ->
+                    ClrRegexLexerOptions(true)
+                | _ ->
+                    ClrRegexLexerOptions.None
+
+            CSharpRegExpBuffer(StringBuffer(text), literalType, lexerOptions) :> _
+
+        override _.DoNotProcessNodeInterior _ = false
+
+        override _.IsPrimaryLanguageApplicable(sourceFile) =
+            sourceFile.LanguageType :? FSharpProjectFileType
+
+        override _.CreateLexerFactory(languageService) =
+            languageService.GetPrimaryLexerFactory()
+
+        override _.AllowsLineBreaks _ = true
+
+        override _.IsWhitespaceToken(token) =
+            token.GetTokenType().IsWhitespace
+
+        // Used only in JS/JSON for now
+        override x.FixValueRangeForLiteral(element) =
+            let startOffset = (x :> IInjectionTargetLanguage).GetStartOffsetForString(element)
+            element.GetTreeTextRange().TrimLeft(startOffset).TrimRight(1)
```

Comment:
Do we need this? We're not going to inject js/json on backend AFAIK, is it needed for something else?

Reply:
I think we can remove this code